samples/cpp/greedy_causal_lm/greedy_causal_lm.cpp

```diff
@@ -4,23 +4,21 @@
 #include "openvino/genai/llm_pipeline.hpp"
 
 int main(int argc, char* argv[]) try {
-    if (3 != argc) {
-        throw std::runtime_error(std::string{"Usage: "} + argv[0] + " <MODEL_DIR> '<PROMPT>'");
-    }
+    if (3 > argc || argc > 4)
+        throw std::runtime_error(std::string{"Usage: "} + argv[0] + " <MODEL_DIR> \"<PROMPT>\" <DEVICE>");
 
     std::string model_path = argv[1];
     std::string prompt = argv[2];
     
     std::string device = "CPU";  // GPU can be used as well
-    ov::genai::LLMPipeline pipe(model_path, device);
+    if (argc > 3) 
+        device = argv[3];
 
+    ov::genai::LLMPipeline pipe(model_path, device);
     ov::genai::GenerationConfig config;
     config.max_new_tokens = 100;
-    auto streamer = [](std::string subword){ std::cout << subword << std::flush; return false; };
-    
-    // Since the streamer is set, the results will
-    // be printed each time a new token is generated.
-    pipe.generate(prompt, config, streamer);
+    std::string result = pipe.generate(prompt, config);
+    std::cout << result;
```

Comment:
But why `DecodedResults` always prints scores? Maybe it's better if i change `operator<<` so that it will not print scores when there is only one sequence?

Reply:
You could have implement is as `std::cout << result.texts.front() << '\n'`. It's not much harder. While difference in string cast may be unexpected.